TestFoundation/TestNSAttributedString.swift

```diff
@@ -0,0 +1,65 @@
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2015 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+
+
+
+#if DEPLOYMENT_RUNTIME_OBJC || os(Linux)
+    import Foundation
+    import XCTest
+#else
+    import SwiftFoundation
+    import SwiftXCTest
+#endif
+
+
+
+class TestNSAttributedString : XCTestCase {
+    
+    static var allTests: [(String, TestNSAttributedString -> () throws -> Void)] {
+        return [
+            ("test_initWithString", test_initWithString),
+            ("test_initWithStringAndAttributes", test_initWithStringAndAttributes)
+        ]
+    }
+    
+    func test_initWithString() {
+        let string = "Lorem 😀 ipsum dolor sit amet, consectetur adipiscing elit. ⌘ Phasellus consectetur et sem vitae consectetur. Nam venenatis lectus a laoreet blandit. ಠ_ರೃ"
+        let attrString = NSAttributedString(string: string)
+        XCTAssertEqual(attrString.string, string)
+        XCTAssertEqual(attrString.length, string.utf16Count)
+        
+        var range = NSRange()
+        let attrs = attrString.attributesAtIndex(0, effectiveRange: &range)
```

Comment:
It seems that this will cause a compiler error with current swift toolchain:

```
let attrs = attrString.attributesAtIndex(0, effectiveRange: nil)
```

But the darwin counterpart can actually accept `nil` as a valid argument.
Not sure if a change is needed in the function signature.